v3/lint/profile.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * ZLint Copyright 2021 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lint
+
+type Profile struct {
+	// Name is a lowercase underscore-separated string describing what a given
+	// lints a given profile aggregates.
+	Name string `json:"name"`
+
+	// A human-readable description of what the Profile checks. Usually copied
+	// directly from the CA/B Baseline Requirements, RFC 5280, or other published
+	// document.
+	Description string `json:"description,omitempty"`
+
+	// The source of the check, e.g. "BRs: 6.1.6" or "RFC 5280: 4.1.2.6".
+	Citation string `json:"citation,omitempty"`
+
+	// Programmatic source of the check, BRs, RFC5280, or ZLint
+	Source LintSource `json:"source,omitempty"`
```

Comment:
I'm wondering whether it's acceptable to consider a `Profile`'s citations/sources as the aggregate of the citations/sources of the lints indicated in `LintNames`. If so maybe we can simplify the model a little bit and drop these two fields.

If you think there is a use-case for having profile-level citations/sources I think the comments need to be adjusted slightly. "source of the check" reads like there's a singular "check" which (IMO) conflicts with the multiple `LintNames`.

Reply:
(Sorry, this is likely an artifact that I hadn't bothered to aggregate the discussion in #583 into #594)

So these profiles aren't arbitrary groupings. The plan is that they _do_ come a from source requirement (the first impetus of which is https://github.com/sleevi/cabforum-docs/pull/36). So the idea is that a parent source lists a collection of requirements with an explanation as to _why_ they are grouped.

```
CABF/BR x.y.z

Section A:
    Due to whatever reason <requirement A>, <requirement B>, <requirement C> from RFRC5280 should...
```

```go
{
   Citation: "Due to whatever reason, these requirements from RFC5280 should...",
   Source: lint.CABFBR_X_Y_Y
}
```

> "source of the check" reads like there's a singular "check"

Effectively there is, in that if one sub-check fails then the whole check fails, and thus you have failed the profile from the cited source. A bit like how a boolean expression can comprised of multiple boolean expressions (and named).

```go
// In Chris' humble opinion the following is what is considered
// altogether a nice day.
func isANiceDay(sunny, cool, butNotTooCool, lightbreeze, bool) {
    return sunny && cool && butNotTooCool && lightbreeze
}
```
Each individual one may-or-may-not be desirable in their own context. However, grouped together they answer the single question, "By Chris' definition is this a nice day?"

Reply:
....although that was my take on it. Since @sleevi is the first source of these, does it make sense for there to be a simple  inheritance type scheme where we can just say that a profile's citations is the list of its lints' citations?